deepspeed/elasticity/elastic_agent.py

```diff
@@ -48,7 +48,7 @@ def _set_master_addr_port(store: Store,
                               master_port: Optional[int],
                               local_addr: Optional[str] = None):
         if master_port is None:
-            sock = _get_socket_with_port()
+            sock = get_free_port()
```

Comment:
Here, you can simply do:
```
if master_port is None:
    master_port = get_free_port()
```

Reply:
I recall there being a reason that we did not do this before, I'll have to look through some other PRs.

Reply:
I can't think of any reason not to do this since `get_free_port` is basically doing this very thing already.